### ISSUE TITLE
Saliency model sources

### ISSUE THREAD

**Issue by esizikova** [wvangansbeke/Unsupervised-Semantic-Segmentation#12]
Could you please clarify how the supervised and unsupervised saliency masks were obtained? Were pre-trained models from DeepUSPS and BAS-Net used? For DeepUSPS, was MSRA-10K or MSRA-B used for training? Thanks!

**Comment by wvangansbeke** [comment]
Hi @esizikova,

To keep it simple, you can use the available pretrained models (from the public repositories for BASNet and DeepUSPS). However, we retrained some of them to double check. For DeepUSPS MSRA-B was used (see DeepUSPS paper).  

Regarding DeepUSPS: You first need to make predictions with DeepUSPS on MSRA10k/MSRA-B (or a similar dataset) and then train BASNet on the obtained pseudo-labels. This bootstrapping procedure results in masks with higher quality. This is also explained on pages 5-6 in section 4.1 of our paper. After training, you need to threshold the output at 0.5 to obtain a binary mask. We additionally filter out images for which the area of the salient object is smaller than 1% of the total area.

**Comment by seyeeet** [comment]
@wvangansbeke I know it is not part of the paper, but would it be possible to share the unsupervised deepusps code/model with the trained weights since they are highly important part of the paper?

**Comment by wvangansbeke** [comment]
Hi @seyeeet,

Yes. Will share the BasNet model.

**Comment by zadaianchuk** [comment]
Hi @wvangansbeke, 
thanks a lot for the really interesting paper! 
The weights of trained BasNet (with unsupervised labels from DeepUSPS) or code would be really helpful! When are you planning to share it? I'm asking to understand do I need to implement it or can I wait for the release of your implementation.

**Comment by wvangansbeke** [comment]
Hi @zadaianchuk,
You can find the model and more explanation [here](https://github.com/wvangansbeke/Unsupervised-Semantic-Segmentation/tree/main/saliency).

**Comment by zadaianchuk** [comment]
Hi @wvangansbeke  
Thanks a lot for sharing the weights of pretrained BasNet! Unfortunately, I anyway need to train BASNet from scratch on DeepUSPS outputs on MSRA10k/MSRA-B. Can you specify the number of epochs you where training BASNet on this dataset? I guess it is only one parameter that is changed from default settings [here ](https://github.com/xuebinqin/BASNet/blob/master/basnet_train.py), if not can you please provide all the hyperparameters that where changed from default values? Also, exact parameters that were used for DeepUSPS would be useful. E.g. can you specify what network was used as a DeepUSPS backbone? Was it pretrained with any other dataset (e.g. CityScapes or ImageNet)? This would make the whole pipeline of unsupervised saliency bootstrapping fully reproducible.
